### PR TITLE
Fix map restrictions content bug

### LINKED ISSUE #356: Fix mobility map createRestrictedArea function bug
Some restrictions content are not being displayed in the map. 
We currently check which restriction with the `===` comparison operator; this is incorrect since some backend restriction names end with a number e.g. `do-not-park0`. 

Let's use `.startsWith` method or a better solution

image

### PR DESCRIPTION
**Pull request by DeeTheDev**
Fixes #356 

- Fix map restrictions content not popping up when clicked

### Popup content images

Screenshot_20230117_110332
Screenshot_20230117_110349
Screenshot_20230117_110341

### CODE REVIEW

File: webapp/src/modules/mapBuilder.js [hunk 1]
```diff
@@ -224,18 +224,16 @@ export default class MapBuilder {
...
       polygonFillOpacity = 0.2;
     }
-    if (restriction === "do-not-ride") {
+    if (restriction.startsWith("do-not-ride")) {
       popup.setContent(ridingRestrictionContent);
       polygonFillOpacity = 0.2;
     }
-    if (restriction === "do-not-park-or-ride") {
-      popup.setContent(allRestrictionsContent);
+    if (restriction.startsWith("do-not-park-or-ride")) {
```
Comment (rgalanakis) [inline webapp/src/modules/mapBuilder.js:235]: Seems like this is pre-empted by line 227. This should be the first condition, and it should be using if/else.

Comment (rgalanakis) [reply webapp/src/modules/mapBuilder.js:235]: (you can copy the polygon fill opacity line a 3rd time, not a big deal)

Comment (DeeTheDev) [reply webapp/src/modules/mapBuilder.js:235]: Thanks for catching that.

Comment (DeeTheDev) [reply webapp/src/modules/mapBuilder.js:235]: The idea for polyfill was for overlapping layers to have a darker tint but it's totally unnecessary, I'll remove it.

File: webapp/src/modules/mapBuilder.js [hunk 2]
```diff
@@ -261,7 +258,7 @@ export default class MapBuilder {
   }
 
   startRefreshTimer(interval, bounds, mcg) {
-    if (this._refreshId && this._ongoingTrip) {
+    if (this._refreshId) {
```
Comment (DeeTheDev) [inline webapp/src/modules/mapBuilder.js:261]: @rgalanakis This variable was probably in the works and wasn't cleaned up when it was removed.

**Pull request merged**